homeassistant/components/mta/config_flow.py

```diff
@@ -0,0 +1,162 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MTAConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for MTA."""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.data: dict[str, Any] = {}
+        self.stops: dict[str, str] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self.data[CONF_LINE] = user_input[CONF_LINE]
+            return await self.async_step_stop()
+
+        lines = sorted(LINE_TO_FEED.keys())
+        line_options = [SelectOptionDict(value=line, label=line) for line in lines]
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_LINE): SelectSelector(
+                        SelectSelectorConfig(
+                            options=line_options,
+                            mode=SelectSelectorMode.DROPDOWN,
+                        )
+                    ),
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_stop(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the stop step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            stop_id = user_input[CONF_STOP_ID]
+            self.data[CONF_STOP_ID] = stop_id
+            stop_name = self.stops.get(stop_id, stop_id)
+            self.data[CONF_STOP_NAME] = stop_name
+
+            unique_id = f"{self.data[CONF_LINE]}_{stop_id}"
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+            try:
+                await self._async_test_connection()
+            except Exception:  # noqa: BLE001
+                errors["base"] = "cannot_connect"
+            else:
+                title = f"{self.data[CONF_LINE]} Line - {stop_name}"
+                return self.async_create_entry(
+                    title=title,
+                    data=self.data,
+                )
+
+        try:
+            self.stops = await self._async_get_stops(self.data[CONF_LINE])
+        except Exception:
+            _LOGGER.exception("Error fetching stops")
+            errors["base"] = "cannot_connect"
+
+        if not self.stops:
+            errors["base"] = "no_stops"
+
+        if errors:
+            return self.async_show_form(
+                step_id="stop",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_STOP_ID): TextSelector(),
+                    }
+                ),
+                errors=errors,
+                description_placeholders={"line": self.data[CONF_LINE]},
+            )
```

Comment:
At this point I think I'd rather abort as this isn't something that the user has caused with a previous setting. By not allowing them to use a free form text, we prevent issues as well